pkg/deploy/deploy.go

```diff
@@ -452,13 +435,19 @@ func performPatch(ctx context.Context, cli client.Client, obj, found *unstructur
 	return cli.Patch(ctx, found, client.RawPatch(types.ApplyPatchType, data), client.ForceOwnership, client.FieldOwner(owner.GetName()))
 }
 
-func updateResource(ctx context.Context, cli client.Client, obj, found *unstructured.Unstructured, owner metav1.Object, componentName string) error {
+func updateResource(ctx context.Context, cli client.Client, res *resource.Resource, found *unstructured.Unstructured, owner metav1.Object) error {
```

Comment:
i think the biggest difference of having componentName and not are:
the old one (with it) can have a fast return from `skipUpdateOnWhitelistedFields`. 
with the new code, removeWhitelistedFromDeployment always gets call on all resources, need go thro the removeWhitelistedFromDeployment and run()
so in reality, it makes all deployment's resources possible to be updated

Reply:
Yes, it is a design decision (and it is a separate patch on top), should it be configured per-component, or not. I can make a map "component -> whitelisted fields", it can be hardcoded similar way like now. But I was also thinking about making it configurable via DSC. It is more dynamic change comparing with demonstrated earlier "replicas" implementation.

Reply:
i think a component map should be better to start with.
if we make it in DSC, it is a new requirement to API, will be difficult unless it is done like a feature, plus for downstream this can involve support cases.

Reply:
Updated but open for suggestions. I feel that it's more clear at least to have string -> strings and keep plugin inside.